packager/media/event/muxer_listener_factory.cc

```diff
@@ -105,14 +105,28 @@ std::unique_ptr<MuxerListener> MuxerListenerFactory::CreateListener(
       combined_listener->AddListener(
           CreateMediaInfoDumpListenerInternal(stream.media_info_output));
     }
-    if (mpd_notifier_) {
-      combined_listener->AddListener(
-          CreateMpdListenerInternal(stream, mpd_notifier_));
-    }
-    if (hls_notifier_) {
-      for (auto& listener :
-           CreateHlsListenersInternal(stream, stream_index, hls_notifier_)) {
-        combined_listener->AddListener(std::move(listener));
+
+    if (stream.dash_only || stream.hls_only) {
```

Comment:
You can simplify the code to:

```suggestion
    if (mpd_notifier_ && !stream.hls_only) {
      ...
    }
    if (hls_notifier_ && !stream.dash_only) {
      ...
    }
```